fs/fs_zenfs.cc

```diff
@@ -255,11 +256,65 @@ ZenFS::~ZenFS() {
   zbd_->LogZoneUsage();
   LogFiles();
 
+  if (gc_worker_ != nullptr) {
+    run_gc_worker_ = false;
+    gc_worker_->join();
+  }
+
   meta_log_.reset(nullptr);
   ClearFiles();
   delete zbd_;
 }
 
+void ZenFS::GCWorker() {
+  while (run_gc_worker_) {
+    uint64_t non_free = zbd_->GetUsedSpace() + zbd_->GetReclaimableSpace();
+    uint64_t free = zbd_->GetFreeSpace();
+    uint64_t free_percent = (100 * free) / (free + non_free);
+    ZenFSSnapshot snapshot;
+    ZenFSSnapshotOptions options;
+
+    usleep(1000 * 1000 * 10);
+
+    if (free_percent > GC_START_LEVEL) continue;
```

Comment:
Maybe we should add another threshold that prevents reclaiming zones if there is not much reclaimable space.

Reply:
The code is already doing that, and it depends on how much space is available on the disk